numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2406,6 +2416,49 @@ def get_function(self, name):
     def get_global_symbol(self, name):
         """Return a MemoryPointer referring to the named symbol"""
 
+    def _set_callables(self, callbacks, attr):
+        if not isinstance(callbacks, list):
+            raise TypeError("callbacks must be a list")
+
+        for f in callbacks:
+            if not callable(f):
+                raise TypeError("callback must be callable")
+
+        setattr(self, attr, callbacks)
+
+    def set_setup_functions(self, callbacks):
+        """Set the setup callback for this module"""
+        self._set_callables(callbacks, "setup_functions")
+
+    def set_teardown_functions(self, callbacks):
+        """Set the finalize callback for this module"""
+        self._set_callables(callbacks, "teardown_functions")
```

Comment:
Do we need all these extra functions instead of just taking setup and teardown functions as parameters to the `__init__()` function?